oemof_visio/energy_system_graph.py

```diff
@@ -0,0 +1,279 @@
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            label,
+            shape="rectangle",
+            fontsize="10",
+            fixedsize="shape",
+            width="4.1",
+            height="0.3",
+            style="filled",
+            color="lightgrey",
+        )
+
+    def add_sink(self, label="Sink", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            fixed_width_text(label, char_num=self.txt_width),
+            shape="trapezium",
+            fontsize=self.txt_fontsize,
+        )
+
+    def add_source(self, label="Source", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            fixed_width_text(label, char_num=self.txt_width),
+            shape="invtrapezium",
+            fontsize=self.txt_fontsize,
+        )
+
+    def add_transformer(self, label="Transformer", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            fixed_width_text(label, char_num=self.txt_width),
+            shape="rectangle",
+            fontsize=self.txt_fontsize,
+        )
+
+    def add_storage(self, label="Storage", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            fixed_width_text(label, char_num=self.txt_width),
+            shape="rectangle",
+            style="rounded",
+            fontsize=self.txt_fontsize,
+        )
+
+    def add_component(self, label="component", subgraph=None):
+        if subgraph is None:
+            dot = self.dot
+        else:
+            dot = subgraph
+        dot.node(
+            fixed_width_text(label, char_num=self.txt_width),
+            fontsize=self.txt_fontsize,
+        )
+
+    def connect(self, a, b):
+        """Draw an arrow from node a to node b
+
+        Parameters
+        ----------
+        a: `oemof.solph.network.Node`
+            An oemof node (usually a Bus or a Component)
+
+        b: `oemof.solph.network.Node`
+            An oemof node (usually a Bus or a Component)
+        """
+        if not isinstance(a, oemof.solph.network.Bus):
+            a = fixed_width_text(a.label, char_num=self.txt_width)
+        else:
+            a = a.label
+        if not isinstance(b, oemof.solph.network.Bus):
+            b = fixed_width_text(b.label, char_num=self.txt_width)
+        else:
+            b = b.label
+
+        self.dot.edge(a, b)
+
+    def view(self, **kwargs):
+        """Call the view method of the DiGraph instance"""
+        self.dot.view(**kwargs)
+
+    def render(self, **kwargs):
+        """Call the render method of the DiGraph instance"""
+        print(self.dot.render(**kwargs))
+
+
+if __name__ == "__main__":
```

Comment:
> Should I also do a PR (at https://github.com/oemof/oemof-examples/tree/master/oemof_examples) that I link to this PR and we merge them conjointly?

Yes.

Reply:
done in https://github.com/oemof/oemof-examples/pull/61